syft/interfaces/protobuf_interface.py

```diff
@@ -0,0 +1,26 @@
+def get_all_subclasses(cls):
+    def check_implementation(s):
+        bufferize_implemented = s.bufferize.__qualname__.startswith(s.__name__)
+        unbufferize_implemented = s.unbufferize.__qualname__.startswith(s.__name__)
+        get_protobuf_schema_implemented = s.get_protobuf_schema.__qualname__.startswith(s.__name__)
+        return bufferize_implemented and unbufferize_implemented and get_protobuf_schema_implemented
+
+    original_subclasses = {s for s in cls.__subclasses__() if check_implementation(s)}
+    sub_sets = {
+        s for c in cls.__subclasses__() for s in get_all_subclasses(c) if check_implementation(s)
+    }
+    return original_subclasses.union(sub_sets)
+
+
+class ProtobufInterface:
```

Comment:
👍 

Can we apply this same class to msgpack serialization with `simplify`/`detail` and call it something like `Serializable`?

Reply:
Sure! I could add a wrapper for libraries as well to have a similar behavior,

Reply:
Libraries?

Reply:
Instead of keeping a list with those objects from libraries (torch, numpy) we can use in msgpack/protobuf, it might be useful to have a wrapper with the same behaviour as this interface? Might be overkill.